kernel/typeops.ml

```diff
@@ -548,22 +548,26 @@ let rec execute env cstr =
     | Construct c ->
       cstr, type_of_constructor env c
 
-    | Case (ci,p,iv,c,lf) ->
+    | Case (ci, u, pms, p, iv, c, lf) ->
+        (** FIXME: change type_of_case to handle the compact form *)
+        let (ci, p, iv, c, lf) = expand_case env (ci, u, pms, p, iv, c, lf) in
```

Comment:
Do you expect the result type of a case to stay as `App (it_mkLambda_or_LetIn pctx p.preturn) (indices ++ [c])` when switching to use the compact form here or will we forego the it_mkLambda, forcing the expansion of lets in pctx ? I can imagine the first representation could be easier to handle during reduction to avoid unnecessary substitutions. Also, it would be backwards compatible.

Reply:
The obvious disadvantage is to have to rebuild `pctx`.

Reply:
No particular preference, I think that starting from here we need to be focused on efficiency while retaining backwards compatibility as much as possible. So, essentially, we need to experiment. The typing rules are under-specified anyways, so we can always decide to switch from one another.